sites/skipthegames_com/ad_parser.py

```diff
@@ -0,0 +1,131 @@
+from typing import List, Dict
+from collections import defaultdict
+from bs4.element import Tag
+from sites.base_ad_parser import BaseAdParser
+from utils.basic import get_matches
+
+
+class AdParser(BaseAdParser):
+    def __init__(self, html):
+        super().__init__(html)
+        self.HTTPS = "https:"
+        self.GENDER_DICT = {"Female escort": "female",
+                            "Male escort": "male",
+                            "TS escort": "trans"}
+        self.info_dict = defaultdict(lambda: "")
+        if self.is_ad():
+            self.info_dict = self._get_info_table()
+
+    def _get_info_table(self) -> Dict:
+        post_body = self.soup.select_one("div#post-body")
+        info_dict = {}
+        if post_body:
+            table = post_body.find("tbody")
+            if table:
+                table_entries = table.find_all("tr")
+                for row in table_entries:
+                    key, value = row.find_all("p")
+                    info_dict[key.text] = value.text.strip()
+        return info_dict
+
+    def is_ad(self) -> bool:
+        ad_indicator = self.soup.select_one("div#post-container")
+        return bool(ad_indicator)
+
+    def primary_phone_number(self) -> str:
+        # TODO: We haven't found an ethical way to get around this site's hCaptcha yet.
+        return ""
+
+    def phone_numbers(self) -> List[str]:
+        prim_phone_number = self.primary_phone_number()
+        phone_numbers_found = {*[prim_phone_number]} if prim_phone_number else set()
+        if self.ad_text():
+            phone_numbers_found.update(get_matches(self.phone_re, self.ad_text()))
+        return list(phone_numbers_found)
+
+    def date_posted(self) -> str:
+        date = self.soup.select_one(".post-timestamp")
+        return date.text.split()[-1]
+
+    def name(self) -> str:
+        title = self.soup.select_one(".post-title")
+        return title.text.strip()
+
+    def primary_email(self) -> str:
+        # TODO: We haven't found an ethical way to get around this site's hCaptcha yet.
+        return None
+
+    # pylint: disable=assignment-from-none
+    def emails(self) -> List[str]:
+        primary_email = self.primary_email()
+        emails_found = {*[primary_email]} if primary_email else set()
+        if self.ad_text():
+            emails_found.update(get_matches(self.email_re, self.ad_text()))
+        return list(emails_found)
+
+    def social(self) -> List[str]:
+        return []
+
+    def age(self) -> str:
+        age = self.soup.select_one("span.ad_display_h3_age")
+        if age:
+            return age.text.split()[0]
+        return ""
+
+    def image_urls(self) -> List[str]:
+        images = self.soup.find_all("img", {"class": "th ctt"})
+        return [self.HTTPS + tag["src"] for tag in images]
+
+    def location(self) -> str:
+        return self.soup.title.text.split(" in ")[-1].strip()
+
+    def ethnicity(self) -> str:
+        return self.info_dict["Race / Regionality"]
```

Comment:
```suggestion
    def ethnicity(self) -> str:
        label = self.soup.select_one(f"td:-soup-contains('Race / Regionality')")
        if label:
            return label.next_sibling.next_sibling.text
        return ""
```

This is the only function that uses info_dict so if we do this we can remove a lot of code above.

Reply:
Beautiful soup's sibling-related functions tend to be wonky and unreliable (tend to mix in a lot of blank elements), I avoid using them. Having a dictionary is cleaner and will be useful if we decide to capture more data later.